server/services/completedAdjudicationsService.ts

```diff
@@ -1,16 +1,58 @@
 import HmppsAuthClient, { User } from '../data/hmppsAuthClient'
 import ManageAdjudicationsClient from '../data/manageAdjudicationsClient'
-import { CompletedAdjudicationSummary } from '../data/CompletedAdjudicationsData'
 import { PageResponse } from '../utils/pageResponse'
 import PageRequest from '../utils/pageRequest'
+import { ReportedAdjudication } from '../data/ReportedAdjudicationResult'
+import PrisonApiClient from '../data/prisonApiClient'
+import { convertToTitleCase, formatTimestampToDate, timestampToDate } from '../utils/utils'
+import PrisonerSimpleResult from '../data/prisonerSimpleResult'
+
+interface ReportedAdjudicationEnhanced extends ReportedAdjudication {
+  displayName: string
+  friendlyName: string
+  formattedDateTimeOfIncident: string
+  dateTimeOfIncident: Date
+}
 
 export default class CompletedAdjudicationsService {
   constructor(private readonly hmppsAuthClient: HmppsAuthClient) {}
 
-  async getCompletedAdjudications(
+  async getYourCompletedAdjudications(
     user: User,
     pageRequest: PageRequest
-  ): Promise<PageResponse<CompletedAdjudicationSummary>> {
-    return new ManageAdjudicationsClient(user.token).getCompletedAdjudications(user, pageRequest)
+  ): Promise<PageResponse<ReportedAdjudicationEnhanced>> {
+    const pageResponse = await new ManageAdjudicationsClient(user.token).getYourCompletedAdjudications(
+      user.activeCaseLoadId,
+      pageRequest
+    )
+    const prisonerDetails = new Map(
+      (
+        await new PrisonApiClient(user.token).getBatchPrisonerDetails(pageResponse.content.map(_ => _.prisonerNumber))
+      ).map(prisonerDetail => [prisonerDetail.offenderNo, prisonerDetail])
+    )
+
+    return pageResponse.map(reportedAdjudication =>
+      this.enhanceReportedAdjudication(reportedAdjudication, prisonerDetails.get(reportedAdjudication.prisonerNumber))
+    )
+  }
+
+  enhanceReportedAdjudication(
+    reportedAdjudication: ReportedAdjudication,
+    prisonerResult: PrisonerSimpleResult
+  ): ReportedAdjudicationEnhanced {
+    const displayName =
+      (prisonerResult && convertToTitleCase(`${prisonerResult.lastName}, ${prisonerResult.firstName}`)) || ''
+    const friendlyName =
+      (prisonerResult && convertToTitleCase(`${prisonerResult.firstName} ${prisonerResult.lastName}`)) || ''
+    return {
+      ...reportedAdjudication,
+      displayName,
+      friendlyName,
+      dateTimeOfIncident: timestampToDate(reportedAdjudication.incidentDetails.dateTimeOfIncident),
```

Comment:
I am not sure there is any need to use this new function to convert to a date, as it is eventually used for sorting.... and i don't think the sorting code (in JS) knows what a Date is

I don't know - it might work (maybe you took this from DPS?) - just checking

Reply:
So - you are correct - its for sorting. The sorting seems to work as expected in the front end